Add Int32 support for NNAPI

axitkhurana opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61025
* #61024
* #61023
* #61022
* #61021
* #59529
* #59434
* **#59365**
* #59364
* #58543
* #60885
* #58541
* #58540
* #58539

Summary:

Support Int32 tensors in NNAPI converter

Test Plan:

Local testing with FB prod models

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881040](https://our.internmc.facebook.com/intern/diff/D28881040)

Commit:
Broken zeros

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

cuspymd commented on torch/backends/_nnapi/serializer.py:
```diff
@@ -869,6 +910,26 @@ def add_tuple_construct(self, node):
+            outputs[0] = self.add_tensor_operand(
+                node.outputsAt(0),
+                in_oper._replace(op_type=NNAPI_OperandCode.TENSOR_INT32))
+        else:
+            raise Exception("Cast only implemented for aten::Int")
```
It might be good to move on to the beginning of the function.

cuspymd commented on torch/backends/_nnapi/serializer.py:
```diff
@@ -869,6 +910,26 @@ def add_tuple_construct(self, node):
+
+        in_id, in_oper = self.get_tensor_operand_by_jitval(node.inputsAt(0))
+
+        inputs = [None] * 1
+        inputs[0] = in_id
```
Is there a reason to assign this way even though only one element?

Commit:
Update on "Broken zeros"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881040).

Commit:
Update on "Broken zeros"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881040](https://our.internmc.facebook.com/intern/diff/D28881040)

[ghstack-poisoned]

Commit:
Update on "Broken zeros"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881040](https://our.internmc.facebook.com/intern/diff/D28881040)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881040).

Commit:
Update on "Add Int32 support for NNAPI"


Summary:

Support Int32 tensors in NNAPI converter

Test Plan:

Local testing with FB prod models

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881040](https://our.internmc.facebook.com/intern/diff/D28881040)

[ghstack-poisoned]

Commit:
Update on "Add Int32 support for NNAPI"


Summary:

Support Int32 tensors in NNAPI converter

Test Plan:

Local testing with FB prod models

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881040](https://our.internmc.facebook.com/intern/diff/D28881040)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881040).

Commit:
Update on "Add Int32 support for NNAPI"


Summary:

Support Int32 tensors in NNAPI converter

Test Plan:

Local testing with FB prod models

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881040](https://our.internmc.facebook.com/intern/diff/D28881040)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881040).

Commit:
Update on "Add Int32 support for NNAPI"


Summary:

Support Int32 tensors in NNAPI converter

Test Plan:

Local testing with FB prod models

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881040](https://our.internmc.facebook.com/intern/diff/D28881040)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881040).

Commit:
Update on "Add Int32 support for NNAPI"


Summary:

Support Int32 tensors in NNAPI converter

Test Plan:

Local testing with FB prod models

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881040](https://our.internmc.facebook.com/intern/diff/D28881040)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881040).

Commit:
Update on "Add Int32 support for NNAPI"


Summary:

Support Int32 tensors in NNAPI converter

Test Plan:

Local testing with FB prod models

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881040](https://our.internmc.facebook.com/intern/diff/D28881040)

[ghstack-poisoned]

Commit:
Update on "Add Int32 support for NNAPI"


Summary:

Support Int32 tensors in NNAPI converter

Test Plan:

Local testing with FB prod models

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881040](https://our.internmc.facebook.com/intern/diff/D28881040)

[ghstack-poisoned]

Commit:
Update on "Add Int32 support for NNAPI"


Summary:

Support Int32 tensors in NNAPI converter

Test Plan:

Local testing with FB prod models

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881040](https://our.internmc.facebook.com/intern/diff/D28881040)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881040).
